models/dataset.go

```diff
@@ -0,0 +1,145 @@
+	UsageNotes    *[]UsageNote         `bson:"usage_notes,omitempty"    json:"usage_notes,omitempty"`
+	Version       int                  `bson:"version,omitempty"        json:"version,omitempty"`
+}
+
+// Alert represents an object containing information on an alert
+type Alert struct {
+	Date        string `bson:"date,omitempty"        json:"date,omitempty"`
+	Description string `bson:"description,omitempty" json:"description,omitempty"`
+	Type        string `bson:"type,omitempty"        json:"type,omitempty"`
+}
+
+// DownloadList represents a list of objects of containing information on the downloadable files
+type DownloadList struct {
+	CSV  *DownloadObject `bson:"csv,omitempty" json:"csv,omitempty"`
+	CSVW *DownloadObject `bson:"csvw,omitempty" json:"csvw,omitempty"`
+	XLS  *DownloadObject `bson:"xls,omitempty" json:"xls,omitempty"`
+}
+
+// DownloadObject represents information on the downloadable file
+type DownloadObject struct {
+	HRef    string `bson:"href,omitempty"  json:"href,omitempty"`
+	Private string `bson:"private,omitempty" json:"private,omitempty"`
+	Public  string `bson:"public,omitempty" json:"public,omitempty"`
+	// TODO size is in bytes and probably should be an int64 instead of a string this
+	// will have to change for several services (filter API, exporter services and web)
+	Size string `bson:"size,omitempty" json:"size,omitempty"`
+}
+
+// LatestChange represents an object contining
+// information on a single change between versions
+type LatestChange struct {
+	Description string `bson:"description,omitempty" json:"description,omitempty"`
+	Name        string `bson:"name,omitempty"        json:"name,omitempty"`
+	Type        string `bson:"type,omitempty"        json:"type,omitempty"`
+}
+
+// TemporalFrequency represents a frequency for a particular period of time
+type TemporalFrequency struct {
+	EndDate   string `bson:"end_date,omitempty"    json:"end_date,omitempty"`
+	Frequency string `bson:"frequency,omitempty"   json:"frequency,omitempty"`
+	StartDate string `bson:"start_date,omitempty"  json:"start_date,omitempty"`
+}
+
+// UsageNote represents a note containing extra information associated to the resource
+type UsageNote struct {
+	Note  string `bson:"note,omitempty"     json:"note,omitempty"`
+	Title string `bson:"title,omitempty"    json:"title,omitempty"`
+}
+
+// VersionLinks represents a list of specific links related to the version resource for an edition of a dataset
+type VersionLinks struct {
+	Dataset    *LinkObject `bson:"dataset,omitempty"     json:"dataset,omitempty"`
+	Dimensions *LinkObject `bson:"dimensions,omitempty"  json:"dimensions,omitempty"`
+	Edition    *LinkObject `bson:"edition,omitempty"     json:"edition,omitempty"`
+	Self       *LinkObject `bson:"self,omitempty"        json:"self,omitempty"`
+	Spatial    *LinkObject `bson:"spatial,omitempty"     json:"spatial,omitempty"`
+	Version    *LinkObject `bson:"version,omitempty"     json:"-"`
+}
```

Comment:
Hmmmm. I am guessing this is now a duplication of the models in the dataset API. Is it worth having a team discussion on how best to handle this? 

1. Duplicating this many models is not optimal and will be easy to fall into a situation where they get out of sync.
2. Having this API depend on the dataset API feels like a bad idea.
3. Having a separate model package that both the Observation and Dataset API feels cleanest but is not without its own headaches.

I'm in favour of 1 or 3. Im not so bothered on which but I think it would be useful to agree as a team which approach we take.